targets/wasi.json

```diff
@@ -0,0 +1,22 @@
+{
+	"llvm-target":   "wasm32--wasi",
+	"build-tags":    ["js", "wasm", "wasi"],
```

Comment:
I would _really_ like to avoid the complexity around the JS/WebAssembly compatibility. So instead of adding the `js` and `wasm` build tags, can you only include the `wasi` build tag instead? It would mean the target is treated more like a baremetal target instead of a WebAssembly target at the source code level.

The main reason is the added complexity around the loader. There is already the issue with overriding the syscall or not overriding it, and I would like to keep those exceptions to an absolute minimum.

For this to work, you may need to add the `wasi` build tag to a few source files in addition to the `baremetal` build tag.

For one example of the complexities of trying to be compatible with `js`/`wasm`:

```go
// TODO: add crypto/rand
```